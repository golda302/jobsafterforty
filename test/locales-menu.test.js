import React from 'react';
import LocalesMenu from '../src/locales-menu';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import darkBaseTheme from 'material-ui/styles/baseThemes/darkBaseTheme';

test('Menu changes the value when an option is selected', () => {
	const aLocales=[
  	{shortText: "he", primaryText: "עברית"},
  	{shortText: "en", primaryText: "English"}];
  	function setLocale(sLocale) {
  		console.log("setting locale to " + sLocale);
  	}
	const component = renderer.create(	
    <MuiThemeProvider muiTheme={getMuiTheme(darkBaseTheme)}>
   <LocalesMenu locales={aLocales} sLocale='he' setLocale={setLocale}/></MuiThemeProvider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});