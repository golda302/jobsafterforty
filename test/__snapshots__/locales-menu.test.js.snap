exports[`test Menu changes the value when an option is selected 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <DropDownMenu
    anchorOrigin={
        Object {
            "horizontal": "left",
            "vertical": "top",
          }
    }
    animated={true}
    autoWidth={true}
    disabled={false}
    iconButton={<NavigationArrowDropDown />}
    maxHeight={500}
    multiple={false}
    onChange={[Function]}
    openImmediately={false}
    value="he">
    <MenuItem
        anchorOrigin={
            Object {
                "horizontal": "right",
                "vertical": "top",
              }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        primaryText="עברית"
        value="he" />
    <MenuItem
        anchorOrigin={
            Object {
                "horizontal": "right",
                "vertical": "top",
              }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        primaryText="English"
        value="en" />
</DropDownMenu>,
  "nodes": Array [
    <DropDownMenu
      anchorOrigin={
            Object {
                  "horizontal": "left",
                  "vertical": "top",
                }
      }
      animated={true}
      autoWidth={true}
      disabled={false}
      iconButton={<NavigationArrowDropDown />}
      maxHeight={500}
      multiple={false}
      onChange={[Function]}
      openImmediately={false}
      value="he">
      <MenuItem
            anchorOrigin={
                  Object {
                        "horizontal": "right",
                        "vertical": "top",
                      }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="עברית"
            value="he" />
      <MenuItem
            anchorOrigin={
                  Object {
                        "horizontal": "right",
                        "vertical": "top",
                      }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="English"
            value="en" />
</DropDownMenu>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <LocalesMenu
        locales={
                Array [
                        Object {
                          "primaryText": "עברית",
                          "shortText": "he",
                        },
                        Object {
                          "primaryText": "English",
                          "shortText": "en",
                        },
                      ]
        }
        sLocale="he"
        setLocamle={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": LocalesMenu {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleChange": [Function],
        "props": Object {
          "locales": Array [
            Object {
              "primaryText": "עברית",
              "shortText": "he",
            },
            Object {
              "primaryText": "English",
              "shortText": "en",
            },
          ],
          "sLocale": "he",
          "setLocamle": [Function],
        },
        "refs": Object {},
        "state": Object {
          "value": "he",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <DropDownMenu
          anchorOrigin={
                    Object {
                              "horizontal": "left",
                              "vertical": "top",
                            }
          }
          animated={true}
          autoWidth={true}
          disabled={false}
          iconButton={<NavigationArrowDropDown />}
          maxHeight={500}
          multiple={false}
          onChange={[Function]}
          openImmediately={false}
          value="he">
          <MenuItem
                    anchorOrigin={
                              Object {
                                        "horizontal": "right",
                                        "vertical": "top",
                                      }
                    }
                    checked={false}
                    desktop={false}
                    disabled={false}
                    focusState="none"
                    insetChildren={false}
                    primaryText="עברית"
                    value="he" />
          <MenuItem
                    anchorOrigin={
                              Object {
                                        "horizontal": "right",
                                        "vertical": "top",
                                      }
                    }
                    checked={false}
                    desktop={false}
                    disabled={false}
                    focusState="none"
                    insetChildren={false}
                    primaryText="English"
                    value="en" />
</DropDownMenu>,
        "_debugID": 2,
        "_renderedOutput": <DropDownMenu
          anchorOrigin={
                    Object {
                              "horizontal": "left",
                              "vertical": "top",
                            }
          }
          animated={true}
          autoWidth={true}
          disabled={false}
          iconButton={<NavigationArrowDropDown />}
          maxHeight={500}
          multiple={false}
          onChange={[Function]}
          openImmediately={false}
          value="he">
          <MenuItem
                    anchorOrigin={
                              Object {
                                        "horizontal": "right",
                                        "vertical": "top",
                                      }
                    }
                    checked={false}
                    desktop={false}
                    disabled={false}
                    focusState="none"
                    insetChildren={false}
                    primaryText="עברית"
                    value="he" />
          <MenuItem
                    anchorOrigin={
                              Object {
                                        "horizontal": "right",
                                        "vertical": "top",
                                      }
                    }
                    checked={false}
                    desktop={false}
                    disabled={false}
                    focusState="none"
                    insetChildren={false}
                    primaryText="English"
                    value="en" />
</DropDownMenu>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LocalesMenu
    locales={
        Array [
            Object {
              "primaryText": "עברית",
              "shortText": "he",
            },
            Object {
              "primaryText": "English",
              "shortText": "en",
            },
          ]
    }
    sLocale="he"
    setLocamle={[Function]} />,
}
`;
